db/migrate/20151119070640_remove_dangling_entries.rb

```diff
@@ -0,0 +1,22 @@
+class RemoveDanglingEntries < ActiveRecord::Migration
+  def up
+    UserSnp.find_each do |us|
+      if Snp.find_by_name(us.snp_name).nil? || Genotype.find_by_id(us.id).nil?
+        UserSnp.destroy(us.id)
+      end
+    end
```

Comment:
Ah, I suppose this works because of the `composite_primary_keys` gem. Nice. :)

I'm all for deleting user-SNPs without genotypes, but I'm not sure we should delete the ones without SNPs, but rather create the SNPs instead.